src/extensions/emulator/optionsHelper.ts

```diff
@@ -46,6 +43,27 @@ export async function buildOptions(options: any): Promise<any> {
   return options;
 }
 
+// Exported for testing
+export async function getParams(options: any, extensionSpec: ExtensionSpec) {
+  const projectId = getProjectId(options, false);
+  const userParams = await paramHelper.readParamsFile(options.testParams);
+  const autoParams = {
+    PROJECT_ID: projectId,
+    EXT_INSTANCE_ID: extensionSpec.name,
+    DATABASE_INSTANCE: projectId,
```

Comment:
Also, is `getFirebaseProjectParams` not appropriate for emulator?

Reply:
Nope, this is actually correct (see google3/firebase/mods/go/dm/instance_converter_test.go or google3/firebase/mods/go/dm/instance_converter.go on the backend). This is just the instance name, which is the first little chunk of the DATABASE_URL. This defaults to projectId.

GetFirebaseProjectParams is not emulator safe, since we need to let you emulate any projectid, real or fake, and that function calls getFirebaseConfig.

Reply:
oh ho. Good to know - thanks for clarifying.